kernel/syscall.c

```diff
@@ -1707,9 +1707,20 @@ long myst_syscall_readlinkat(
     {
         ret = myst_syscall_readlink(pathname, buf, bufsiz);
     }
+    else if (*pathname == '\0')
+    {
+        myst_fdtable_t* fdtable = myst_fdtable_current();
+        myst_fs_t* fs;
+        myst_file_t* file;
+
+        ECHECK(myst_fdtable_get_file(fdtable, dirfd, &fs, &file));
```

Comment:
We should expand `myst_get_absolute_path_from_dirfd` for the benefit of other `SYS_xxxat` syscall handlers.

Reply:
This invoke to `myst_fdtable_get_file` has been removed. The `*pathname='\0'` case requires special treatment in `readlinkat`.